### ISSUE TITLE
xlearn silently crashes and restarts my jupyter nb session when running minimal example

### ISSUE THREAD

**Issue by k8tems** [PKU-Cloud-Lab/xLearn#375]
```python
import xlearn as xl
model = xl.create_fm()
model.setTrain('train.txt')
param = {'task': 'binary', 'lr': .2, 'lambda': 0.002}
model.fit(param, 'model.out')
```
My environment: paperspace gradient A6000 instance(i.e. a docker container)

**Comment by k8tems** [comment]
```cpp
#4  0x00007ffff7cbf7f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x00007ffff766dfbe in __gnu_cxx::__verbose_terminate_handler () at ../../../../libstdc++-v3/libsupc++/vterminate.cc:95
#6  0x00007ffff766c502 in __cxxabiv1::__terminate (handler=<optimized out>) at ../../../../libstdc++-v3/libsupc++/eh_terminate.cc:48
#7  0x00007ffff7666303 in std::terminate () at ../../../../libstdc++-v3/libsupc++/eh_terminate.cc:58
#8  0x00007ffff766c722 in __cxxabiv1::__cxa_throw (obj=0x179a270, tinfo=0x7ffff7789238 <typeinfo for std::logic_error>, 
    dest=0x7ffff76773ac <std::logic_error::~logic_error()>) at ../../../../libstdc++-v3/libsupc++/eh_throw.cc:98
#9  0x00007ffff766646b in std::__throw_logic_error (__s=0x7ffff780d1a0 "basic_string::_M_construct null not valid")
    at ../../../../../libstdc++-v3/src/c++11/functexcept.cc:70
#10 0x00007ffff780259c in get_user_name[abi:cxx11]() () from /opt/conda/lib/python3.10/site-packages/xlearn/libxlearn_api.so
#11 0x00007ffff780284c in get_log_file(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
   from /opt/conda/lib/python3.10/site-packages/xlearn/libxlearn_api.so
#12 0x00007ffff7803c29 in xLearn::Solver::init_log() () from /opt/conda/lib/python3.10/site-packages/xlearn/libxlearn_api.so
#13 0x00007ffff780a7c4 in xLearn::Solver::Initialize(xLearn::HyperParam&) () from /opt/conda/lib/python3.10/site-packages/xlearn/libxlearn_api.so
#14 0x00007ffff77c1586 in XLearnFit () from /opt/conda/lib/python3.10/site-packages/xlearn/libxlearn_api.so
#15 0x00007ffff78e8052 in ffi_call_unix64 () from /opt/conda/lib/python3.10/lib-dynload/../../libffi.so.8
#16 0x00007ffff78e6925 in ffi_call_int () from /opt/conda/lib/python3.10/lib-dynload/../../libffi.so.8
#17 0x00007ffff78e706e in ffi_call () from /opt/conda/lib/python3.10/lib-dynload/../../libffi.so.8
#18 0x00007ffff78f81e7 in _call_function_pointer (argtypecount=<optimized out>, argcount=2, resmem=0x7fffffffce10, restype=<optimized out>, 
    atypes=<optimized out>, avalues=<optimized out>, pProc=0x7ffff77c1470 <XLearnFit>, flags=4353)
    at /usr/local/src/conda/python-3.10.14/Modules/_ctypes/callproc.c:916
#19 _ctypes_callproc (pProc=0x7ffff77c1470 <XLearnFit>, argtuple=0x7ffff7ae9240, flags=4353, argtypes=<optimized out>, restype=0x818ba0, checker=0x0)
    at /usr/local/src/conda/python-3.10.14/Modules/_ctypes/callproc.c:1262
#20 0x00007ffff78f769b in PyCFuncPtr_call (self=<optimized out>, inargs=<optimized out>, kwds=0x0)
    at /usr/local/src/conda/python-3.10.14/Modules/_ctypes/_ctypes.c:4221
#21 0x00000000004f705b in _PyObject_MakeTpCall (tstate=0x777220, callable=0x7fffcd6c68c0, args=<optimized out>, nargs=<optimized out>, keywords=0x0)
    at /usr/local/src/conda/python-3.10.14/Objects/call.c:215
#22 0x00000000004f3106 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x7fffe9266d90, callable=0x7fffcd6c68c0, 
    tstate=<optimized out>) at /usr/local/src/conda/python-3.10.14/Include/cpython/abstract.h:112
#23 _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x7fffe9266d90, callable=0x7fffcd6c68c0, tstate=<optimized out>)
    at /usr/local/src/conda/python-3.10.14/Include/cpython/abstract.h:99
#24 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x7fffe9266d90, callable=0x7fffcd6c68c0)
    at /usr/local/src/conda/python-3.10.14/Include/cpython/abstract.h:123
#25 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffd120, tstate=<optimized out>)
    at /usr/local/src/conda/python-3.10.14/Python/ceval.c:5893
#26 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=0x7fffe9266c00, throwflag=<optimized out>) at /usr/local/src/conda/python-3.10.14/Python/ceval.c:4181
#27 0x00000000004fdd4f in _PyEval_EvalFrame (throwflag=0, f=0x7fffe9266c00, tstate=0x777220)
    at /usr/local/src/conda/python-3.10.14/Include/internal/pycore_ceval.h:46
--Type <RET> for more, q to quit, c to continue without paging--q
Quit
```
According to the call stack info above, it seems the `get_user_name` function is the direct cause of the crash.

https://github.com/aksnzhy/xlearn/blob/4c240aa0aa63c1d105fb9aec583adc2ad2840368/src/base/system.h#L48-L51

Populating the `USER` environment variable fixed the problem.

**Comment by k8tems** [comment]
Forgot to mention:
my initial snippet does not include the mandatory `model.setValidation('valid.txt')` which was causing another exception(crash).
